pkg/model/requests_processing_test.go

```diff
@@ -68,24 +69,71 @@ func testRequestsProcessing(t *testing.T, describe spec.G, it spec.S) {
 			assert.Equal(t, simulator.StockName("RequestsProcessing [99]"), subject.Name())
 		})
 	})
-
 	describe("Add()", func() {
-		var request simulator.Entity
+		var request RequestEntity
 
-		it.Before(func() {
-			bufferStock := NewRequestsRoutingStock(envFake, NewReplicasActiveStock(), nil)
-			request = NewRequestEntity(envFake, bufferStock, RequestConfig{CPUTimeMillis: 200, IOTimeMillis: 200, Timeout: 3 * time.Second})
-			subject.Add(request)
+		describe("there is no free cpu resource", func() {
+			it.Before(func() {
+				*rawSubject.occupiedCPUCapacityMillisPerSecond = *rawSubject.totalCPUCapacityMillisPerSecond
+				routingStock := NewRequestsRoutingStock(envFake, NewReplicasActiveStock(), nil)
+				request = NewRequestEntity(envFake, routingStock, RequestConfig{CPUTimeMillis: 200, IOTimeMillis: 200, Timeout: 3 * time.Second})
+				subject.Add(request)
+			})
+			describe("scheduling processing", func() {
+				it("schedules a movement from RequestsProcessing to RequestsFailed", func() {
+					assert.Equal(t, simulator.StockName("RequestsFailed"), envFake.Movements[0].To().Name())
+				})
+			})
+
+			it("there is 1 entity in processingStock", func() {
+				assert.Equal(t, subject.Count(), uint64(1))
+			})
 		})
+		describe("request fails as request total time exceeds request timeout", func() {
+			it.Before(func() {
+				routingStock := NewRequestsRoutingStock(envFake, NewReplicasActiveStock(), nil)
+				request = NewRequestEntity(envFake, routingStock, RequestConfig{CPUTimeMillis: 20000, IOTimeMillis: 200, Timeout: 3 * time.Second})
+				subject.Add(request)
+			})
+
+			it("increments the number of requests since last Stat()", func() {
+				assert.Equal(t, int32(1), rawSubject.numRequestsSinceLast)
```

Comment:
How is this relevant to tested component, is this part of responsibility?

In general: The tested component has certain public interface, responsibility, "the contract" with other code. Try to test that public contract, not the internals of how exactly the results are achieved. I.e. test externally visible state, using public interface. 

E.g.  what's the external result of the timeout? I guess, we should test that the request failed, and that cpuUtilization was recalculated correctly.

See also https://testing.googleblog.com/2015/01/testing-on-toilet-prefer-testing-public.html

Reply:
I totally agree with you. Thank you for the link. I've read it. Very useful resource. Originally it wasn't my test. I was thinking whether delete it or stay alive :) Fixed it (see commit 2d6552efb24f1c1eed205483f07717a4798bf296).